### PR TITLE
🐛🚨🚨 Source Pinterest: fix for access_token authentication and OAuth2 authentication 

### PR DESCRIPTION
**Pull request by xander1235**
## What
*Fixes* [#19235](https://github.com/airbytehq/airbyte/issues/19235)
*Describe what the change is solving*
Solving the Pinterest source connection with access_token. 

*It helps to add screenshots if it affects the frontend.*
[https://airbytehq.slack.com/archives/C027KKE4BCZ/p1697811485984309](https://airbytehq.slack.com/archives/C027KKE4BCZ/p1697811485984309)
## How
*Describe the solution*
Differentiated the authentication based on the auth_method.
If the auth_token is access_token, Basic auth is handled.
If the auth_token is OAuth2, OAuth validation is done.

## Recommended reading order
1. `source.py`
2. `Dockerfile`

## 🚨 User Impact 🚨
Yes

*Are there any breaking changes? What is the end result perceived by the user?*
If the access_token as a authentication type is selected, the flow can not be moved forward for the source. User won't be able to use this source with access_token type authentication.

*For connector PRs, use this section to explain which type of semantic versioning bump occurs as a result of the changes. Refer to our [Semantic Versioning for Connectors](https://docs.airbyte.com/contributing-to-airbyte/#semantic-versioning-for-connectors) guidelines for more information. **Breaking changes to connectors must be documented by an Airbyte engineer (PR author, or reviewer for community PRs) by using the [Breaking Change Release Playbook](https://docs.google.com/document/d/1VYQggHbL_PN0dDDu7rCyzBLGRtX-R3cpwXaY8QxEgzw/edit).***

*If there are breaking changes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.*
Please consider this PR for fixing breaking changes 🚨🚨.

## Pre-merge Actions
*Expand the relevant checklist and delete the others.*

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter
Community memeber

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Build is successful

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-pinterest/source_pinterest/source.py [hunk 1]
```diff
@@ -74,18 +74,35 @@ def get_authenticator(config):
...
-        authenticator = self.get_authenticator(config)
+
+        config = config.get("credentials") or config
+
         url = f"{PinterestStream.url_base}user_account"
-        auth_headers = {"Accept": "application/json", **authenticator.get_auth_header()}
+        auth_headers = {}
         try:
+            authenticator = any
+            if config.get("auth_method") == "access_token":
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-pinterest/source_pinterest/source.py:84]: Why we compare with `access_token`. In spec we have only `oauth2.0` auth_method value

**Review by lazebnyi — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-pinterest/source_pinterest/source.py (hunk 1 above)
Comment (marcosmarxm) [reply airbyte-integrations/connectors/source-pinterest/source_pinterest/source.py:84]: The contribution was made before https://github.com/airbytehq/airbyte/pull/31876 where Pinterest had two different auth methods.

Comment (lazebnyi) [reply airbyte-integrations/connectors/source-pinterest/source_pinterest/source.py:84]: so I believe now it is not relevant? or we need change condition

**Comment by marcosmarxm** [comment]
I'm closing this contribution because Pinterest doesn't support `access_token` auth method. This is based on the previous work made in #31876 

> Research showed that our code never supported "access_token" based authentication.
I've tried to implement it but "access_token" provided via pinterest UI is:
a short-term token (expires in 1 day) and
is limited to 3 streams only (has limited scope).
So I decided that this method should be removed at all.

**Pull request closed without merging**